src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -346,12 +353,18 @@ public Symbol WellKnownMember(WellKnownMember wm, bool isOptional = false)
                 throw new MissingPredefinedMember(diagnostic);
             }
 
+            // NOTE: The result is not null when isOptional is false, but there is no annotation to specify that.
```

Comment:
If this can be solved with MemberNotNull when it becomes available, please add link to [https://github.com/dotnet/roslyn/issues/41964](https://github.com/dotnet/roslyn/issues/41964) here

Reply:
I am splitting this into two methods for optional and not optional in the next iteration.

---
In reply to: [386586467](https://github.com/dotnet/roslyn/pull/42068#discussion_r386586467) [](ancestors = 386586467)